flutter_vlc_player_platform_interface/lib/src/utils/options/vlc_subtitle_options.dart

```diff
@@ -0,0 +1,99 @@
+import '../../enums/subtitles/vlc_subtitle_text_direction.dart';
+import '../../enums/subtitles/vlc_subtitle_color.dart';
+import '../../enums/subtitles/vlc_subtitle_thickness.dart';
+
+class VlcSubtitleOptions {
+  final List<String> options;
+
+  VlcSubtitleOptions(this.options);
+
+  /// Set subtitle font (must be accessible by vlc)
+  static String font(String font) {
+    return '--freetype-font=$font';
+  }
+
+  /// Set subtitle mono font (must be accessible by vlc)
+  static String monofont(String monofont) {
+    return '--freetype-monofont=$monofont';
+  }
+
+  /// Set subtitle font size
+  static String fontSize(int size) {
+    return '--freetype-fontsize=$size';
+  }
+
+  /// Set subtitle rel font size
+  static String relFontSize(int size) {
+    return '--freetype-rel-fontsize=$size';
+  }
+
+  /// Enable/Disable subtitle bold style
+  static String boldStyle(bool enable) {
+    return enable ? '--freetype-bold' : '--no-freetype-bold';
+  }
+
+  /// Set subtitle opacity [0 .. 255]
+  static String opacity(int opacity) {
+    return '--freetype-opacity=$opacity';
+  }
+
+  /// Set subtitle color
+  static String color(VlcSubtitleColor color) {
+    return '--freetype-color=${color.value}';
+  }
+
+  /// Set subtitle text direction {0 (Left to right), 1 (Right to left), 2 (Auto)}
+  static String textDirection(VlcSubtitleTextDirection direction) {
+    return '--freetype-text-direction=${direction.value}';
+  }
+
+  /// Set subtitle background opacity [0 .. 255]
+  static String backgroundOpacity(int opacity) {
+    return '--freetype-background-opacity=$opacity';
+  }
+
+  /// Set subtitle background color
+  static String backgroundColor(VlcSubtitleColor color) {
+    return '--freetype-background-color=${color.value}';
+  }
+
+  /// Set subtitle outline opacity [0 .. 255]
+  static String outlineOpacity(int opacity) {
+    return '--freetype-outline-opacity=$opacity';
+  }
+
+  /// Set subtitle outline color
+  static String outlineColor(VlcSubtitleColor color) {
```

Comment:
I like that it's a bit more structured than just an `int` input, but the current implementation limits the user to a predefined set of colors.

We should probably provide a way to create custom colors. We could do that by turning a `VlcSubtitleColor` enum into an ADT, with a public constructor for custom colors, at the same time keeping the predefined values as `static const` fields on the class.
```dart
@immutable
class VlcSubtitleColor {
  final int value;

  static const navy = VlcSubtitleColor(12345);

  const VlcSubtitleColor(this.value);
  const VlcSubtitleColor.rgb({
    int red = 0,
    int green = 0,
    int blue = 0,
  }): value = ...;
}
```